src/components/SearchForm/SearchForm.js

```diff
@@ -1,38 +1,46 @@
-function SearchForm({ handleSearchClick }) {
-  return (
-    <form className="search-form">
-      <h1 className="search-form__title">Whats going on in the world</h1>
-      <h2 className="search-form__subtitle">
-        Find the latest news on any topic and save them in your personal
-        account.
-      </h2>
-      <div className="search-form__field-wrapper">
-        <input
-          className="search-form__field"
-          placeholder="Enter a Topic"
-        ></input>
-        <button
-          type="submit"
-          className="search-form__button"
-          onClick={handleSearchClick}
-        >
-          Search
-        </button>
-      </div>
-      {/* <input
-        type="email"
-        placeholder="Enter your email address"
-        class="footer__form-input"
-      />
-      <button class="footer__form-button button button_type_dark">
-        Join
-        <img
-          src="./images/tiny-arrow.png"
-          alt="arrow icon"
-          class="footer__button-icon"
-        />
-      </button> */}
-    </form>
-  );
-}
-export default SearchForm;
+import newsApi from "../../utils/NewsApi";
+
+function SearchForm({
+  handleArticlesSearch,
+  searchValue,
+  setSearchValue,
+  handleSearchClick,
+}) {
+  async function handleSubmit(e) {
+    e.preventDefault();
+
+    if (searchValue.length > 0) {
+      handleSearchClick();
+      try {
+        const res = await newsApi.getArticles(searchValue);
+        if (res.articles && res.articles.length > 0) {
+          handleArticlesSearch(res.articles);
+        } else {
+          handleArticlesSearch(null);
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    }
+  }
+  return (
+    <form className="search-form" action="submit" onSubmit={handleSubmit}>
+      <h1 className="search-form__title">Whats going on in the world</h1>
+      <h2 className="search-form__subtitle">
+        Find the latest news on any topic and save them in your personal
+        account.
+      </h2>
+      <div className="search-form__field-wrapper">
+        <input
+          className="search-form__field"
+          placeholder="Enter a Topic"
+          onChange={(e) => setSearchValue(e.target.value)}

```

Comment:
If you have `onChange` or ` onInput` in an input then you need to add attribute `value` to this input.
This is how React controls inputs.
To read more about **controlled components**: https://reactjs.org/docs/forms.html#controlled-components